07-routing-gossip.md

```diff
@@ -0,0 +1,252 @@
++--------------------------------------+----+---------------------+
+```
+
+## The `channel_update` message
+
+After a channel has been initially announced, each side independently
+announces its fees and minimum expiry for HTLCs.  It uses the 8-byte
+channel shortid which matches the `channel_announcement` and one byte
+to indicate which end this is.  It can do this multiple times, if
+it wants to change fees.
+
+1. type: 258 (`MSG_CHANNEL_UPDATE`)
+2. data:
+	* [64:signature]
+	* [4:blockheight]
+	* [3:blockindex]
+	* [1:outputindex]
+	* [4:timestamp]
+	* [1:side]
+	* [1:pad]
+    * [2:expiry]
+	* [4:htlc-minimum-msat]
+	* [4:fee-base-msat]
+	* [4:fee-proportional-millionths]
+
+### Requirements
+
+The creating node MUST set `signature` to the signature of the
+double-SHA of the entire remaining packet after `signature` using its own `node-id`.
+
+The creating node MUST set `blockheight`, `blockindex` and `outputindex` to
+match those in the already-sent `channel_announcement` message, and MUST set `side` to 0 if the creating node is `node-id-1` in that message, otherwise 1.
+
+The creating node MUST set `timestamp` to greater than zero, and MUST set it to greater than any previously-sent `channel_update` for this channel, and MUST set `pad` to zero.
+
+It MUST set `expiry` to the number of blocks it will subtract from an incoming HTLC's `expiry`.  It MUST set `htlc-minimum-msat` to the minimum HTLC value it will accept, in millisatoshi.  It MUST set `fee-base-msat` to the base fee it will charge for any HTLC, in millisatoshi, and `fee-proportional-millionths` to the amount it will charge per millionth of a satoshi.
+
+The node SHOULD accept HTLCs which pay a fee equal or greater than:
+
+	fee-base-msat + htlc-amount-msat * fee-proportional-millionths / 1000000
+
+The receiving node SHOULD fail the connection if `side` is not 0 or 1.
+It MUST ignore the contents of `pad`.  The receiving node SHOULD fail
+the connection if `signature` is invalid or incorrect for the entire
+message including unknown fields following `signature`, and MUST NOT
+further process the message.  The receiving node SHOULD ignore `ipv6`
+if `port` is zero.  It SHOULD ignore the message if `blockheight`,
+`blockindex` and `outputindex` do not correspond to a previously
+known, unspent channel from `channel_announcement` or if `timestamp`
+is not greater than than the last-received `channel_announcement` for
+this channel and `side`.  Otherwise, if the `timestamp` is equal to
+the last-received `channel_announcement` and the fields other than
+`signature` differ, the node MAY blacklist this node-id and forget all
+channels associated with it.  Otherwise the receiving node SHOULD
+queue the message for rebroadcasting.
+
+## Rebroadcasting
+
+Nodes receiving an announcement verify that this is the latest update from the announcing node by comparing the included timestamp and update their local view of the network's topology accordingly.
```

Comment:
I think you mean that the receiving node should keep the latest update per message type and per channel right ?

Which means for announcing node A:
- latest `node_announcement` message with `node-id`=A if any
- all `channel_announcement` messages with `node-id-1`=A OR `node-id-2`=A not previously broadcast
- latest `channel_update` for each unique  `blockheight` \ `blockindex` \ `outputindex`